core/federated/RTI/rti_lib.c

```diff
@@ -566,7 +568,10 @@ void update_federate_next_event_tag_locked(uint16_t federate_id, tag_t next_even
         next_event_tag = min_in_transit_tag;
     }
 
-    _RTI.federates[federate_id].next_event = next_event_tag;
+    if (lf_tag_compare(_RTI.federates[federate_id].next_event, next_event_tag) != 0) {
+        _RTI.federates[federate_id].next_event = next_event_tag;
+        //FIXME: notify upstream federates
```

Comment:
Maybe separate this change into another PR rather than putting a FIXME here in this PR?

Reply:
I moved this change into another draft PR ([reactor-c/pull/176](https://github.com/lf-lang/reactor-c/pull/176)).
Also, removed this change at this commit (67f9d53de775c85665eb6e3da01da9921016a2ea).